python/ray/tune/trial.py

```diff
@@ -97,11 +97,11 @@ def __init__(
 
         # Immutable config
         self.trainable_name = trainable_name
-        self.config = config
+        self.config = config or {}
         self.local_dir = local_dir
         self.experiment_tag = experiment_tag
         self.resources = resources
-        self.stopping_criterion = stopping_criterion
+        self.stopping_criterion = stopping_criterion or {}
```

Comment:
could this be an error? We have a loop

```python
for k in stopping_criterion:
```

above, but `stopping_criterion` may be None.

Reply:
yeah it will fail - I've pushed a fix.

Reply:
This isn't done right?

Reply:
Fixed